vaadin-text-field-flow-parent/vaadin-text-field-flow/src/main/java/com/vaadin/flow/component/textfield/AbstractNumberField.java

```diff
@@ -388,15 +414,185 @@ private boolean isValidByStep(T value) {
+     * Note: updating the object content that is gotten from this method will
+     * not update the lang on the component if not set back using
+     * {@link AbstractNumberField#setI18n(AbstractNumberFieldI18n)}
+     *
+     * @return the i18n object. It will be {@code null}, If the i18n properties
+     *         weren't set.
+     */
+    protected AbstractNumberFieldI18n getI18n() {
+        return i18n;
+    }
+
+    /**
+     * Sets the internationalization properties for this component.
+     *
+     * @param i18n
+     *            the internationalized properties, not {@code null}
+     */
+    protected void setI18n(AbstractNumberFieldI18n i18n) {
+        this.i18n = Objects.requireNonNull(i18n,
+                "The i18n properties object should not be null");
+    }
+
+    private String getBadInputErrorMessage() {
+        return Optional.ofNullable(i18n)
+                .map(AbstractNumberFieldI18n::getBadInputErrorMessage)
+                .orElse("");
+    }
+
+    private String getRequiredErrorMessage() {
+        return Optional.ofNullable(i18n)
+                .map(AbstractNumberFieldI18n::getRequiredErrorMessage)
+                .orElse("");
+    }
+
+    private String getMinErrorMessage() {
+        return Optional.ofNullable(i18n)
+                .map(AbstractNumberFieldI18n::getMinErrorMessage).orElse("");
+    }
+
+    private String getMaxErrorMessage() {
+        return Optional.ofNullable(i18n)
+                .map(AbstractNumberFieldI18n::getMaxErrorMessage).orElse("");
+    }
+
+    private String getStepErrorMessage() {
+        return Optional.ofNullable(i18n)
+                .map(AbstractNumberFieldI18n::getStepErrorMessage).orElse("");
+    }
+
+    /**
+     * The abstract internationalization properties for
+     * {@link AbstractNumberField}.
+     */
+    public static abstract class AbstractNumberFieldI18n
+            implements Serializable {
+        /**
+         * Gets the error message displayed when the field contains user input
+         * that the server is unable to convert to a {@link Number}.
+         *
+         * @return the error message or {@code null} if not set
+         */
+        public abstract String getBadInputErrorMessage();
+
+        /**
+         * Sets the error message to display when the field contains user input
+         * that the server is unable to convert to a {@link Number}.
+         *
+         * @param errorMessage
+         *            the error message to set, or {@code null} to clear
+         * @return this instance for method chaining
+         */
+        public abstract AbstractNumberFieldI18n setBadInputErrorMessage(
+                String errorMessage);
+
+        /**
+         * Gets the error message displayed when the field is required but
+         * empty.
+         *
+         * @return the error message or {@code null} if not set
+         * @see AbstractNumberField#isRequired()
+         * @see AbstractNumberField#setRequired(boolean)
```

Comment:
**question:** Should we mention `setRequiredIndicatorVisible` and `isRequiredIndicatorVisible` here as well?